app/models/feed.rb

```diff
@@ -47,10 +58,46 @@ def default_runtime_settings?
   end
 
   def published_url
-    if default?
-      "#{podcast.base_published_url}/#{file_name}"
-    else
-      "#{podcast.base_published_url}/#{slug}/#{file_name}"
+    "#{podcast.base_published_url}/#{published_path}"
+  end
+
+  def published_path
+    default? ? file_name : "#{slug}/#{file_name}"
+  end
+
+  def feed_episodes
+    include_in_feed = []
+    feed_max = display_episodes_count.to_i
+    filtered_episodes.each do |ep|
+      next unless include_episode_categories?(ep)
+  
+      include_in_feed << ep if ep.include_in_feed?
+      break if (feed_max > 0) && (include_in_feed.size >= feed_max)
     end
+    include_in_feed
+  end
+
+  def include_episode_categories?(ep)
+    return true if (include_tags || []).length <= 0
+    tags = include_tags.map { |cat| normalize_category(cat) }
+    cats = (ep || []).categories.map { |cat| normalize_category(cat) }
+    (tags & cats).length > 0
+  end
+
+  def normalize_category(cat)
+    cat.to_s.downcase.gsub(/[^ a-z0-9_-]/, '').gsub(/\s+/, ' ').strip
+  end
+
+  def filtered_episodes
+    podcast.episodes.published_by(episode_offset_seconds.to_i)
+  end
+
+  def enclosure_template
+    self[:enclosure_template] || Feed.enclosure_template_default
```

Comment:
Okay, so if I'm reading this all right ...

In the migration, you copy the podcast enclosure-template over too the feed(s).  So these will all be non-nil for default_feeds when this deploys.

But any newly added Feeds (and we should fixup the TTD adfree demo feed) will get the new enclosure template, with the additional fanciness.